test/e2e/node/pods.go

```diff
@@ -425,7 +427,7 @@ var _ = SIGDescribe("Pods Extended", func() {
 							if !hasTerminalPhase {
 								var names []string
 								for _, status := range pod.Status.ContainerStatuses {
-									if status.State.Terminated != nil || status.State.Running != nil {
+									if status.State.Running != nil {
```

Comment:
I think this was just wrong before, but we never hit it because we were failing to sync out terminated container statuses. Now that we're doing that correctly, a Pending pod with a terminated container (perhaps because it was deleted before it reached Running) would cause this test to fail.

Reply:
Note: this line came from https://github.com/kubernetes/kubernetes/pull/88440 which explicitly says in the description "The kubelet will consider any container without the `waiting` or `terminated` state to be still running, like it does for other operations." which doesn't match the code here.